.github/workflows/regression-check.yml

```diff
@@ -0,0 +1,91 @@
+name: Regression checking
+
+on:
+  workflow_dispatch:
+    inputs:
+      benchmarks:
+        description: '--benchmarks'
+        required: false
+        default: ""
+      fields:
+        description: '--fields'
+        required: false
+        default: "allocated"
+      diff-cutoff-percent:
+        description: '--diff-cutoff-percent'
+        required: false
+        default: 50
+
+# References:
+# https://docs.github.com/en/actions/managing-workflow-runs/manually-running-a-workflow
+
+jobs:
+  check-regressions:
+    env:
+      BENCH_SH: bin/bench.sh
+      BENCHMARKS: ${{ github.event.inputs.benchmarks }}
+      FIELDS: ${{ github.event.inputs.fields }}
+      DIFF_CUTOFF_PERCENT: ${{ github.event.inputs.diff-cutoff-percent }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+
+    - name: Setup haskell
+      uses: actions/setup-haskell@v1
+      with:
+        ghc-version: 8.8.3
+        cabal-version: 3.2
+
+    # Caching isn't currently supported for the workflow_dispatch event.
+    # The following block is essentually dumb and will result in a warning.
+    - uses: actions/cache@v1
+      name: Cache ~/.cabal and bench-report
+      with:
+        path: |
+            ~/.cabal
+            ./benchmark/bench-report/bin/bench-report
+        # Bump the key version to clear the cache
+        key: cache-v1
+
+    # -----------------------------------------------------------------
+    # -- Generate reports for the base branch and upload
+    # -----------------------------------------------------------------
+
+    - name: Checkout the base branch
+      uses: actions/checkout@v2
+      with:
+        ref: master
+
+    - name: Run benchmarks
+      run: |
+        chmod +x $BENCH_SH
+        $BENCH_SH --benchmarks "$BENCHMARKS" --raw --quick
+
+    # -----------------------------------------------------------------
+    # -- Download, generate reports for the current branch and append
+    # -----------------------------------------------------------------
+
+    - name: Checkout the current branch
+      uses: actions/checkout@v2
+      with:
+        clean: false
+
+    - name: Run benchmarks and append
+      run: |
+        chmod +x $BENCH_SH
+        $BENCH_SH --benchmarks "$BENCHMARKS" --raw --quick --append
```

Comment:
Perhaps do not need to use `--quick` as there is not much difference. We can check how much we gain with this.